context.go

```diff
@@ -61,20 +62,19 @@ func (context *Context) Dial(url string, origin string) (*Sender, error) {
 	return sender, nil
 }
 
+type WSHandshake func(*websocket.Config, *http.Request) error
+
 // Creates a WebSocket handshake handler
-func handshake(config *websocket.Config, rq *http.Request) error {
-	ok := false
-	for _, proto := range config.Protocol {
-		if proto == blipProtocolName {
-			ok = true
-			break
+func (context *Context) WebSocketHandshake() WSHandshake {
+	return func(config *websocket.Config, rq *http.Request) error {
+		protocolHeader := rq.Header.Get("Sec-WS-Protocols")
+		if !includesProtocol(protocolHeader, WebSocketProtocolName) {
+			context.log("Error: Client doesn't support WS protocol %s, only %s", WebSocketProtocolName, protocolHeader)
```

Comment:
I tested this by trying to connect with a couchbase-lite client that's currently using the master branch of couchbase-lite-ios, and got this in the sync gateway logs:

```
2017-12-18T17:59:46.013-08:00 BLIP: Error: Client doesn't support WS protocol BLIP_3a2, only 

```

It looks like the older couchbase-lite-ios client doesn't send the `Sec-WS-Protocols` header.  Here are the headers I'm seeing:

![screen shot 2017-12-18 at 5 59 38 pm](https://user-images.githubusercontent.com/296876/34137023-7e887af2-e41d-11e7-8ccc-209cf07e7151.png)

I don't know if it's a huge deal, since cblite 2 has never gone GA yet, but I just thought I'd mention this observation.